src/Promitor.Core.Scraping/Configuration/Serialization/IErrorReporter.cs

```diff
@@ -0,0 +1,24 @@
+﻿using YamlDotNet.RepresentationModel;
+
+namespace Promitor.Core.Scraping.Configuration.Serialization
+{
+    /// <summary>
+    /// Used to report errors discovered during the deserialization process.
+    /// </summary>
+    public interface IErrorReporter
```

Comment:
I'm thinking out loud - Since it doesn't only handle errors, should we stick with the current name?

From what we're doing here, should we rename it to IValidationReporter or IDeserializationReporter?

Reply:
Although current approach is fine actually

Reply:
I'm happy to rename it - I just wasn't able to think of a better name at the time. The trouble is you've given me two options, and I can't decide between them now 😸 If you don't mind I'll rename it separately - I've got a few more PRs lined up, and it'll make my life easier if I can rename it at the end.

Reply:
That's fine for me! Let's keep it as is.